src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -95,6 +112,19 @@ def run_cfg(self):
     def run_cfg(self, value):
         self._run_cfg = value
 
+    @property
+    def hyperparameters(self):
+        return self._hyperparameters
+
+    @hyperparameters.setter
+    def hyperparameters(self,
+                        value: ty.Dict[str, ty.Union[int, npt.ArrayLike]]):
+        self._hyperparameters = value
+
+    @property
+    def last_run_report(self):
+        return self._report
+
     def solve(self,
               timeout: int,
               target_cost: int = 0,

```

Comment:
Should the target_cost be 0? Below, the default in _update_report is None, and _validate_cost will actually throw an error for 0 cost (which it shouldn't, as this is a valid input).

Additionally, is it possible to use target_cost=None and let the solver run for as many time steps as provided? If so, that should be stated in the docstring.